src/bucket/BucketManagerImpl.cpp

```diff
@@ -867,4 +872,115 @@ BucketManagerImpl::isShutdown() const
 {
     return mIsShutdown;
 }
+
+// Loads a single bucket worth of entries into `map`, deleting dead entries and
+// inserting live or init entries. Should be called in a loop over a BL, from
+// old to new.
+static void
+loadEntriesFromBucket(std::shared_ptr<Bucket> b,
+                      std::map<LedgerKey, LedgerEntry>& map)
+{
+    using namespace std::chrono;
+    medida::Timer timer;
+    BucketInputIterator in(b);
+    timer.Time([&]() {
+        while (in)
+        {
+            BucketEntry const& e = *in;
+            if (e.type() == LIVEENTRY || e.type() == INITENTRY)
+            {
+                map[LedgerEntryKey(e.liveEntry())] = e.liveEntry();
+            }
+            else
+            {
+                if (e.type() != DEADENTRY)
+                {
+                    std::string err = "Malformed bucket: unexpected "
+                                      "non-INIT/LIVE/DEAD entry.";
+                    CLOG_ERROR(Bucket, "{}", err);
+                    throw std::runtime_error(err);
+                }
+                size_t erased = map.erase(e.deadEntry());
+                if (erased != 1)
+                {
+                    std::string err =
+                        fmt::format("DEADENTRY does not exist in ledger: {}",
+                                    xdr::xdr_to_string(e.deadEntry(), "entry"));
+                    CLOG_ERROR(Bucket, "{}", err);
+                    throw std::runtime_error(err);
+                }
+            }
+            ++in;
+        }
+    });
+    nanoseconds ns =
+        timer.duration_unit() * static_cast<nanoseconds::rep>(timer.max());
+    milliseconds ms = duration_cast<milliseconds>(ns);
+    size_t bytesPerSec = (b->getSize() * 1000 / (1 + ms.count()));
+    CLOG_INFO(Bucket, "Read {}-byte bucket file in {}ms ({}/s)", b->getSize(),
```

Comment:
I _think_ that fmt::format now has a built-in way to convert durations

Reply:
Fixed.